pkg/log/pipeline_reader.go

```diff
@@ -194,6 +191,28 @@ func (r *Reader) setUpTask(taskNumber int, tr trh.Run) {
 	r.setTask(tr.Task)
 }
 
+// getOrderedTasks get Tasks in order from Spec.PipelineRef or Spec.PipelineSpec
+// and return trh.Run after converted taskruns into trh.Run.
+func (r *Reader) getOrderedTasks(pr *v1alpha1.PipelineRun) ([]trh.Run, error) {
+	var tasks []v1alpha1.PipelineTask
+
+	switch {
+	case pr.Spec.PipelineRef != nil:
+		pl, err := r.clients.Tekton.TektonV1alpha1().Pipelines(r.ns).Get(pr.Spec.PipelineRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		tasks = pl.Spec.Tasks
+	case pr.Spec.PipelineSpec != nil:
+		tasks = pr.Spec.PipelineSpec.Tasks
+	default:
+		return nil, fmt.Errorf("pipelineRun %s not providing PipelineRef or PipelineSpec", pr.Name)
```

Comment:
nit: fmt.Errorf("PipelineRun %s did not provide PipelineRef or PipelineSpec", pr.Name)

Reply:
Thank for review !
It's based on this document :pray:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
Yes, my fault, agree on the error not being capitalized, but could you change to `fmt.Errorf("pipelinerun %s did not provide PipelineRef or PipelineSpec", pr.Name)` then?

Reply:
Oops, sorry I overlooked that...
Fixed it !!